omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -733,40 +515,56 @@ def _grasp(self, obj):
         grasp_poses = get_grasp_poses_for_object_sticky(obj)
         grasp_pose, object_direction = random.choice(grasp_poses)
 
+        grasp_offset_in_z = self.robot.finger_lengths[self.arm] + m.GRASP_APPROACH_DISTANCE
+
+        # Adjust grasp pose with reset orientation and finger length offset
+        reset_orientation = self._get_reset_eef_pose("world")[self.arm][1]
+        grasp_pos = grasp_pose[0] - object_direction * grasp_offset_in_z
```

Comment:
hmm does `grasp_pose` initially contain this offset and you are undoing it here? Not sure if this is the cleanest code, happy to discuss in case I misunderstand!